test/broadcastClient.ts

```diff
@@ -1,35 +1,45 @@
 import { assert } from "chai";
 import _ from "lodash";
 
+import { ServerInfoResponse } from "../src";
+
 import responses from "./fixtures/responses";
 import rippled from "./fixtures/rippled";
-import setupClient from "./setupClient";
+import { setupBroadcast, teardownClient } from "./setupClient";
 import { ignoreWebSocketDisconnect } from "./testUtils";
 
 const TIMEOUT = 20000;
 
-function checkResult(expected, response) {
+async function checkResult(
+  expected: Record<string, unknown>,
+  response: Record<string, unknown>
+): Promise<Record<string, unknown>> {
   if (expected.txJSON) {
     assert(response.txJSON);
-    assert.deepEqual(JSON.parse(response.txJSON), JSON.parse(expected.txJSON));
+    assert.deepEqual(
+      JSON.parse(response.txJSON as string),
+      JSON.parse(expected.txJSON as string)
+    );
   }
   assert.deepEqual(_.omit(response, "txJSON"), _.omit(expected, "txJSON"));
   return response;
 }
 
 describe("BroadcastClient", function () {
   this.timeout(TIMEOUT);
-  beforeEach(setupClient.setupBroadcast);
-  afterEach(setupClient.teardown);
+  beforeEach(setupBroadcast);
+  afterEach(teardownClient);
 
-  it("base", function () {
+  it("base", async function () {
     this.mocks.forEach((mock) => {
       mock.addResponse("server_info", rippled.server_info.normal);
     });
     assert(this.client.isConnected());
-    return this.client.request({ command: "server_info" }).then((response) => {
-      return checkResult(responses.getServerInfo, response.result.info);
-    });
+    this.client
+      .request({ command: "server_info" })
+      .then(async (response: ServerInfoResponse) => {
+        checkResult(responses.getServerInfo, response.result.info);
+      });
```

Comment:
if you're using the `then` syntax there is no reason to mark this function as `async` (the other one that is. The inner one also doesn't need to because you're not `await`ing it or doing anything with the promise